src/convertToObject.js

```diff
@@ -6,7 +6,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((str) => str.trim())
+    .filter((str) => str.length > 0)
+    .map((str) => str.split(':').map((part) => part.trim()))
+    .reduce((obj, [key, value]) => {
+      obj[key] = value;
+
+      return obj;
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy of the accumulator object in the `reduce` method before each operation. The `reduce` method does not mutate the original array but instead works on the accumulator provided, which in this case is an object (`{}`).